pilot/pkg/model/push_context.go

```diff
@@ -708,6 +718,7 @@ func (ps *PushContext) InitSidecarScopes(env *Environment) error {
 	ps.sidecarsByNamespace = make(map[string][]*SidecarScope)
 	for _, sidecarConfig := range sidecarConfigs {
 		// TODO: add entries with workloadSelectors first before adding namespace-wide entries
+		sidecarConfig := sidecarConfig
```

Comment:
what does this do?

Reply:
Tricky - without the first sidecar ends up in the map ( some optimization, don't remember ).
The change introduces a new variable, and then the map is as expected.

( my test has sidecar in multiple namespaces - it's easy to reproduce ).